index.js

```diff
@@ -177,3 +221,45 @@ function findBestSubtitleFile(list) {
     .filter((x) => x.subs)
     .sort((x, y) => x.priority > y.priority)[0];
 }
+
+function processFormats(formats) {
+  const filteredFormats = formats.filter(({ format_id, fps, height, vcodec }) => format_id !== 'source' && !format_id.startsWith('http') && vcodec && vcodec !== 'none' && !vcodec.includes('av01') && !vcodec.includes('vp9') && !vcodec.includes('vp09') && ((height >= 1080 && fps <= 30) || (height < 1080)))
```

Comment:
I honestly wonder if these booleans can be grouped into helper functions so it's more clear what we're checking. E.g. one for video codec, one for resolution/frame rate, etc...